### PR TITLE
commit 31.03.2019 hw

### PR DESCRIPTION
**Pull request by MateAcademy**

### CODE REVIEW

File: src/main/java/mate/academy/hw4/myArrayList/SimpleArray.java [hunk 1]
```diff
@@ -0,0 +1,58 @@
+package mate.academy.hw4.myArrayList;
+
+import java.util.Iterator;
+
+public class SimpleArray<E> implements Simple<E> {
+    private E[] values;
+
+    public SimpleArray() {
+        values = (E[]) new Object[0];
```
Comment (Siril-O) [inline src/main/java/mate/academy/hw4/myArrayList/SimpleArray.java:9]: Why zero initial size ?

File: src/main/java/mate/academy/hw4/myArrayList/SimpleArray.java [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+    @Override
+    public boolean add(E e) {
+        try {
+            E[] temp = values;
+            values = (E[]) new Object[temp.length + 1];
+            System.arraycopy(temp, 0, values, 0, temp.length);
+            values[values.length - 1] = e;
+            return true;
+        } catch (ClassCastException ex) {
```
Comment (Siril-O) [inline src/main/java/mate/academy/hw4/myArrayList/SimpleArray.java:20]: Why you expect ClassCastException here ?

File: src/main/java/mate/academy/hw4/myArrayList/SimpleArray.java [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+
+    @Override
+    public void delete(int index) {
+        try {
+            E[] temp = values;
+            values = (E[]) new Object[temp.length - 1];
+            System.arraycopy(temp, 0, values, 0, index);
+            int amountEmemAfterIndex = temp.length - index - 1;
+            System.arraycopy(temp, index + 1, values, index, amountEmemAfterIndex);
+        } catch (ClassCastException ex) {
```
Comment (Siril-O) [inline src/main/java/mate/academy/hw4/myArrayList/SimpleArray.java:34]: Why you expect ClassCastException here ?

File: src/main/java/mate/academy/hw4/myArrayList/SimpleArray.java [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+            ex.printStackTrace();
```
Comment (Siril-O) [inline src/main/java/mate/academy/hw4/myArrayList/SimpleArray.java:35]: do not printStackTrace in Util classes, any exceptions should be handled by user of class

**Review by Siril-O — CHANGES_REQUESTED** [review]

File: src/main/java/mate/academy/hw4/myArrayList/SimpleArray.java (hunk 1 above)
Comment (MateAcademy) [reply src/main/java/mate/academy/hw4/myArrayList/SimpleArray.java:9]: In the constructor, I initialized the value, the array should be empty

File: src/main/java/mate/academy/hw4/myArrayList/SimpleArray.java (hunk 2 above)
Comment (MateAcademy) [reply src/main/java/mate/academy/hw4/myArrayList/SimpleArray.java:20]: an exception  is possible in this operation: values = (E[]) new Object[temp.length + 1];

File: src/main/java/mate/academy/hw4/myArrayList/SimpleArray.java (hunk 3 above)
Comment (MateAcademy) [reply src/main/java/mate/academy/hw4/myArrayList/SimpleArray.java:34]: an exception is possible in this operation: values = (E[]) new Object[temp.length - 1];

File: src/main/java/mate/academy/hw4/myArrayList/SimpleArray.java (hunk 4 above)
Comment (MateAcademy) [reply src/main/java/mate/academy/hw4/myArrayList/SimpleArray.java:35]: ok

**Pull request merged**